### ISSUE TITLE
Splitting context and question for BERT

### ISSUE THREAD

**Issue by vivek1410patel** [google-research/bert#495]
The paper suggests that the softmax for start and end logits is computed only over the context hidden states. However, in the code, I could not figure out any place where the hidden states were being split into query and context hidden states. Does that mean that in the code, we are calculating the softmax over the query and context hiddens, and expect the model to learn that the answer lies in context hiddens? Or can you point me to a place, where you are splitting the hidden states and then using the final linear layer only over the context.

**Comment by libertatis** [comment]
You can refer to the function 'write_predictions' in ['run_squad.py'](https://github.com/google-research/bert/blob/master/run_squad.py#L741).
Paying attention to the line [#L787](https://github.com/google-research/bert/blob/master/run_squad.py#L787), it said "We could hypothetically create invalid predictions, e.g., predict that the start of the span is in the question. We throw out all invalid predictions." So the author firstly get n best indexes for start and end, and then just throw out the invalid index in the question. So there is no split operation.
